src/plugins/embeddable/public/lib/panel/embeddable_panel_error.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { EuiPanel } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ErrorLike } from '@kbn/expressions-plugin/common';
+import { distinctUntilChanged, merge, of, switchMap } from 'rxjs';
+import { EditPanelAction } from '../actions';
+import { ErrorEmbeddable, IEmbeddable } from '../embeddables';
+
+interface EmbeddablePanelErrorProps {
+  editPanelAction?: EditPanelAction;
+  embeddable: IEmbeddable;
+  error: ErrorLike;
+}
+
+export function EmbeddablePanelError({
+  editPanelAction,
+  embeddable,
+  error,
+}: EmbeddablePanelErrorProps) {
+  const [isEditable, setEditable] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+  const handleErrorClick = useMemo(
+    () => (isEditable ? () => editPanelAction?.execute({ embeddable }) : undefined),
+    [editPanelAction, embeddable, isEditable]
+  );
+
+  const title = embeddable.getTitle();
+  const actionDisplayName = useMemo(
+    () => editPanelAction?.getDisplayName({ embeddable }),
+    [editPanelAction, embeddable]
+  );
+  const ariaLabel = useMemo(
+    () =>
+      !title
+        ? actionDisplayName
+        : i18n.translate('embeddableApi.panel.editPanel.displayName', {
+            defaultMessage: 'Edit {value}',
+            values: { value: title },
+          }),
+    [title, actionDisplayName]
+  );
+
+  useEffect(() => {
+    const subscription = merge(embeddable.getInput$(), embeddable.getOutput$())
+      .pipe(
+        switchMap(() => editPanelAction?.isCompatible({ embeddable }) ?? of(false)),
+        distinctUntilChanged()
+      )
+      .subscribe(setEditable);
+
+    return () => subscription.unsubscribe();
+  }, [editPanelAction, embeddable]);
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+
+    if (embeddable.renderError) {
+      return embeddable.renderError(ref.current, error);
+    }
+
+    const errorEmbeddable = new ErrorEmbeddable(error, { id: embeddable.id });
+    errorEmbeddable.render(ref.current);
+
+    return () => errorEmbeddable.destroy();
+  }, [embeddable, error]);
+
+  return (
+    <EuiPanel
+      element="div"
+      className="embPanel__content embPanel__content--error"
+      color="transparent"
+      paddingSize="none"
+      data-test-subj="embeddableError"
+      panelRef={ref}
+      role={isEditable ? 'button' : undefined}
+      aria-label={isEditable ? ariaLabel : undefined}
+      onClick={handleErrorClick}
+    />
```

Comment:
Rather than making a clickable `EuiPanel`, would it be better to use an `EuiEmptyPrompt` with an `EuiButton` in the `actions` prop (see [examples in EUI docs](https://elastic.github.io/eui/#/display/empty-prompt))?  I imagine that a dedicated button that notifies the user that they will be taken to edit the erroneous visualization would be a more user-friendly experience. Thoughts?

Reply:
@MichaelMarcialis I've checked the component and the example. I don't think it is applicable in our case:
- The panel here is just a dummy element we use as a placeholder for the UI -- embeddable widgets may render their custom error component. So that we cannot place a button here as it will not be shown in most of the cases.
- This change introduces a generic click handler and doesn't touch any UI.
- The UI present in the video comes from the Visualizations plugin, and it's made in a way to look similar to the error message from the Lens plugin.
- If we would like to change the way it looks, we should create a separate issue that will request updates of the error appearance in three different places at the same time: Lens, Visualizations, and generic Embeddables.

Reply:
Thanks for that explanation, @dokmic. Just to be sure I'm understanding, you're saying that `EmbeddablePanelError` is a placeholder that can be swapped out with a custom error component, correct? Is the swapping out with a custom error component required? Or is it optional, leaving the possibility for this "placeholder" error message to show instead? 

If the swapping out of the `EmbeddablePanelError` component is optional (leaving the possibility for this "placeholder" error panel to be shown to users), I'm inclined to maintain my suggestion that we should instead utilize `EuiEmptyPrompt` here (with a conditional `EuiButton` that takes users to edit the visualization). Visually, I believe `EuiEmptyPrompt` can be styled to look the same as the error message shown in your example (for the sake of consistency with other error messages), just with the addition of the button for better usability.

Let me know if that makes sense or if I'm misunderstanding anything. CCing @flash1293, in case he has any thoughts.

Reply:
Not holding this opinion strongly but I expected to use the context menu for this - click the cogwheel and select the edit action from there.

Reply:
@MichaelMarcialis 
> `EmbeddablePanelError` is a placeholder that can be swapped out with a custom error component, correct? Is the swapping out with a custom error component required? 

Yes, that's correct. It is optional but is overridden in majority of the places. 

> Or is it optional, leaving the possibility for this "placeholder" error message to show instead?
I've checked the code once again, and the change seems impossible at the moment without a significant refactoring. The problem is that we have some internal service to open the panel for edit. It would be tricky to initialize that from Lens or Visualizations. The best solution is to get rid of the custom error components that look similar and use the fallback. But that should be done separately as it requires more work than the original issue and involves more parties.

Is that okay if I create a separate ticket to address that?